src/mesh.cpp

```diff
@@ -102,6 +154,73 @@ StructuredMesh::bin_label(int bin) const {
   }
 }
 
+//==============================================================================
+// Unstructured Mesh implementation
+//==============================================================================
+
+UnstructuredMesh::UnstructuredMesh(pugi::xml_node node) : Mesh(node) {
+  n_dimension_ = 3;
+
+  // check the mesh type
+  if (check_for_node(node, "type")) {
+    auto temp = get_node_value(node, "type", true, true);
+    if (temp != "unstructured") {
+      fatal_error(fmt::format("Invalid mesh type: {}", temp));
+    }
+  }
+
+  // get the filename of the unstructured mesh to load
+  if (check_for_node(node, "filename")) {
+    filename_ = get_node_value(node, "filename");
+    if (!file_exists(filename_)) {
+      fatal_error("Mesh file '" + filename_ + "' does not exist!");
+    }
+  }
+  else {
+    fatal_error(fmt::format("No filename supplied for unstructured mesh with ID: {}", id_));
+  }
+
+  // check if mesh tally data should be written with
+  // statepoint files
+  if (check_for_node(node, "output")) {
+    output_ = get_node_value_bool(node, "output");
+  }
+
+}
+
+void
+UnstructuredMesh::surface_bins_crossed(const Particle& p,
+                                       std::vector<int>& bins) const {
+  throw std::runtime_error{"Unstructured mesh surface tallies are not implemented."};
+}
+
+std::string
+UnstructuredMesh::bin_label(int bin) const {
+  return fmt::format("Mesh Index ({})", bin);
+};
+
+void
+UnstructuredMesh::to_hdf5(hid_t group) const
+{
+    hid_t mesh_group = create_group(group, fmt::format("mesh {}", id_));
+
+    write_dataset(mesh_group, "type", "unstructured");
+    write_dataset(mesh_group, "filename", filename_);
+    write_dataset(mesh_group, "library", this->library());
+    // write volume of each tet
```

Comment:
This is really minor, but since this supports any element type in libMesh, I'd remove the names having to do with tets here, and maybe just replace "tet" by "elem". Could be less confusing for someone reading this

Reply:
Ah, yeah. Great point. I'll update that.